ilastik/experimental/api/_base.py

```diff
@@ -0,0 +1,96 @@
+import vigra
+import numpy
+
+from lazyflow.graph import Graph
+from lazyflow.operators.classifierOperators import OpClassifierPredict
+from lazyflow.operators import OpReorderAxes
+from ilastik.applets.featureSelection.opFeatureSelection import OpFeatureSelection
+from ilastik.experimental import parser
+from .types import Pipeline
+
+
+def from_project_file(path) -> Pipeline:
+    project: parser.PixelClassificationProject
+
+    with parser.IlastikProject(path, "r") as project:
+        if not all([project.data_info, project.features, project.classifier]):
+            raise ValueError("not sufficient data in project file for predition")
+
+        feature_matrix = project.features.as_matrix()
+        classifer = project.classifier
+        num_channels = project.data_info.num_channels
+        axis_order = project.data_info.axis_order
+        num_spatial_dims = len(project.data_info.spatial_axes)
+
+    class _PipelineImpl(Pipeline):
+        def __init__(self):
+            graph = Graph()
+            self._reorder_op = OpReorderAxes(graph=graph, AxisOrder=axis_order)
+
+            self._feature_sel_op = OpFeatureSelection(graph=graph)
+            self._feature_sel_op.InputImage.connect(self._reorder_op.Output)
+            self._feature_sel_op.FeatureIds.setValue(feature_matrix.names)
+            self._feature_sel_op.Scales.setValue(feature_matrix.scales)
+            self._feature_sel_op.SelectionMatrix.setValue(feature_matrix.selections)
+
+            self._predict_op = OpClassifierPredict(graph=graph)
+            self._predict_op.Classifier.setValue(classifer.instance)
+            self._predict_op.Classifier.meta.classifier_factory = classifer.factory
+            self._predict_op.Image.connect(self._feature_sel_op.OutputImage)
+            self._predict_op.LabelsCount.setValue(classifer.label_count)
+
+        def predict(self, data):
+            data = _make_vigra_with_cannel_axis(data)
+            num_channels_in_data = data.shape[data.axistags.index("c")]
+            if num_channels_in_data != num_channels:
+                raise ValueError(
+                    f"Number of channels mismatch. Classifier trained for {num_channels} but input has {num_channels_in_data}"
+                )
+
+            num_spatial_in_data = sum(a.isSpatial() for a in data.axistags)
+            if num_spatial_in_data != num_spatial_dims:
+                raise ValueError(
+                    "Number of spatial dims doesn't match. "
+                    f"Classifier trained for {num_spatial_dims} but input has {num_spatial_in_data}"
+                )
+
+            self._reorder_op.Input.setValue(data)
+
+            return self._predict_op.PMaps.value[...]
+
+    return _PipelineImpl()
+
+
+def _guess_channel_axis_idx(shape):
+    for idx, val in enumerate(shape):
+        if val < 4:
+            return idx
+    return None
+
+
+def _guess_axistags(shape):
```

Comment:
there is `get_default_axisordering` in `lazyflow.helpers`. I'd feel more comfortable not to introduce new code for axis-guessing, especially as this PR already implements a separate code path for pixel classification

Reply:
I added a docstring to this function, the problem with `get_default_axisordering` that it doesn't try to guess the channel axis position. So if I had shape `(1, 128, 128)` then `get_default_axisordering` would guess `zyx` while this function would guess `cyx`

Reply:
okay, how do you know it's `cyx` in that case and not in fact a single `z`-slice. Reading this, and I am very sorry to sort of open a discussion on this, it looks like we should _not_ allow such ambiguity and force users to always supply axistags if they want to predict with bare numpy arrays. I think this will solve a lot of issues in advance and make usage way more clear.

so something like
```python
test_res = classifier.predict(test_arr, axisorder="cyx")
```
I don't think users will hate us for this. After all we do not handle arbitrary matrices, but we have images, that always come with dimensional meanings.

Reply:
Good point, I will use default axis tag guessing and if the array provided has the capability to store tags (e.g. vigra) then use metadata provided. I use singledispatch so it would be easy to add functionality for xarray, ndstructs